src/main/java/pooh/Deadline.java

```diff
@@ -0,0 +1,110 @@
+package pooh;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a deadline task that extends the Task class.
+ * <p>
+ * The Deadline class is a subclass of the Task class, and it adds two additional attributes:
+ * deadlineString and deadlineDateTime to represent the deadline of the task.
+ * </p>
+ */
+public class Deadline extends Task {
+    private final String deadlineString;
+    private final LocalDate deadlineDateTime;
+
+    /**
+     * Constructs a new Deadline object with a description and a deadline.
+     * <p>
+     * The constructor initializes a Deadline object with a description and a deadline,
+     * and it also converts the deadline string into a LocalDate object using the parseDateTime method.
+     * </p>
+     *
+     * @param description    The description of the deadline task.
+     * @param deadlineString The string representation of the deadline.
+     */
+    public Deadline(String description, String deadlineString) {
+        super(description);
+        this.deadlineString = deadlineString;
+        this.deadlineDateTime = parseDateTime(deadlineString);
+    }
+
+    /**
+     * Writes the task details into a string format suitable for file storage.
+     * <p>
+     * This function prepares a string that captures the essential attributes of the Deadline object,
+     * such as its type (D for Deadline), its completion status, its description, and its deadline.
+     * The attributes are separated by a pipe character (|), making it easier to read them back from the file later.
+     * </p>
+     *
+     * @return A string containing the task's details in a format suitable for file storage.
+     */
+    public String writeTaskToFile() {
+        return String.format("%s | %s | %s | %s", "D",
```

Comment:
Is there any reason you chose to use String.format rather than a String concatenation here? Perhaps, it might be clearer that way? Still good use of linebreaks to make the code readable! 👍